cmd/libs/go2idl/go-to-protobuf/protobuf/parser.go

```diff
@@ -97,3 +106,123 @@ func dropExistingTypeDeclarations(decl ast.Decl, existsFn func(string) bool) boo
 	}
 	return false
 }
+
+func RewriteTypesWithProtobufStructTags(name string, structTags map[string]map[string]string) error {
+	fset := token.NewFileSet()
+	src, err := ioutil.ReadFile(name)
+	if err != nil {
+		return err
+	}
+	file, err := parser.ParseFile(fset, name, src, parser.DeclarationErrors|parser.ParseComments)
+	if err != nil {
+		return err
+	}
+
+	allErrs := []error{}
+
+	// set any new struct tags
+	for _, d := range file.Decls {
+		if errs := updateStructTags(d, structTags, []string{"protobuf"}); len(errs) > 0 {
+			allErrs = append(allErrs, errs...)
+		}
+	}
+
+	if len(allErrs) > 0 {
+		var s string
+		for _, err := range allErrs {
+			s += err.Error() + "\n"
+		}
+		return errors.New(s)
+	}
+
+	b := &bytes.Buffer{}
+	if err := printer.Fprint(b, fset, file); err != nil {
+		return err
+	}
+
+	body, err := format.Source(b.Bytes())
+	if err != nil {
+		return fmt.Errorf("%s\n---\nunable to format %q: %v", b, name, err)
+	}
+
+	f, err := os.OpenFile(name, os.O_WRONLY|os.O_TRUNC, 0644)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	if _, err := f.Write(body); err != nil {
+		return err
+	}
+	return f.Close()
+}
+
+func updateStructTags(decl ast.Decl, structTags map[string]map[string]string, toCopy []string) []error {
+	var errs []error
+	switch t := decl.(type) {
```

Comment:
We can save a couple indent layers if we just return if you're not a type definition.